rustls/src/client/tls13.rs

```diff
@@ -786,20 +786,12 @@ fn emit_certificate_tls13(
     auth_context: Option<Vec<u8>>,
     common: &mut CommonState,
 ) {
-    let context = auth_context.unwrap_or_default();
-
-    let mut cert_payload = CertificatePayloadTls13 {
-        context: PayloadU8::new(context),
-        entries: Vec::new(),
+    let mut cert_payload = if let Some(certkey) = certkey {
```

Comment:
Nit: suggest using a `match` here instead of `if let`. I guess we can't hoist out the `new()` call because the iterators have different types?

Reply:
actually, i can hoist out the subject of the iterator, into `certkey.map(|ck| ck.cert.as_ref()).unwrap_or(&[][..]);`